assembl/static2/js/app/main.jsx

```diff
@@ -37,17 +128,59 @@ class Main extends React.Component<Props> {
     );
     return (
       <div className="main">
-        <Navbar location={this.props.location.pathname} themeId={themeId} />
+        <Navbar location={location.pathname} themeId={themeId} />
         <div className="app-content">{children}</div>
         <CookiesBar />
-        <Footer />
+        <Footer
+          hasLegalNotice={hasLegalNotice}
+          hasTermsAndConditions={hasTermsAndConditions}
+          hasCookiesPolicy={hasCookiesPolicy}
+          hasPrivacyPolicy={hasPrivacyPolicy}
+          hasUserGuidelines={hasUserGuidelines}
+        />
       </div>
     );
   }
 }
 
-const mapStateToProps = (state: { timeline: Timeline }) => ({
-  timeline: state.timeline
+const mapStateToProps = state => ({
+  timeline: state.timeline,
+  lang: state.i18n.locale,
+  id: state.context.connectedUserIdBase64
 });
 
-export default connect(mapStateToProps)(Main);
\ No newline at end of file
+const withData = graphql(TabsConditionQuery, {
+  props: ({ data: { hasLegalNotice, hasTermsAndConditions, hasCookiesPolicy, hasPrivacyPolicy, hasUserGuidelines } }) => ({
+    hasLegalNotice: hasLegalNotice,
+    hasTermsAndConditions: hasTermsAndConditions,
+    hasCookiesPolicy: hasCookiesPolicy,
+    hasPrivacyPolicy: hasPrivacyPolicy,
+    hasUserGuidelines: hasUserGuidelines
+  })
+});
+
+export default compose(
+  connect(mapStateToProps),
+  withData,
+  graphql(updateAcceptedCookies, {
+    name: 'updateAcceptedCookies'
+  }),
+  graphql(acceptedCookiesQuery, {
+    skip: props => !props.id,
+    props: ({ data }) => {
+      if (data.error || data.loading) {
+        return {
+          error: data.error,
+          loading: data.loading
+        };
+      }
+
+      return {
+        error: data.error,
+        loading: data.loading,
+        acceptedLegalContentList: data.user.acceptedCookies
+      };
+    }
+  }),
+  manageErrorAndLoading({ displayLoader: false })
+)(Main);
```

Comment:
you knew that it would happen: waaaay too much new code in `main.jsx` :scream: 
we should find another way to do this. Maybe `react-router` could help here. Haven't made a lot of investigation but take a look at this discussion for example: https://stackoverflow.com/questions/28795085/react-router-modal-only-routes
@vincentfretin any ideas?